packages/features/auth/lib/next-auth-options.ts

```diff
@@ -924,7 +928,55 @@ export const AUTH_OPTIONS: AuthOptions = {
       return baseUrl;
     },
   },
-};
+  events: {
+    async signIn(message) {
+      /* only run this code if:
+         - it's a hosted cal account
+         - DUB_API_KEY is configured
+         - it's a new user
+      */
+      const user = message.user as User & {
+        username: string;
+        createdAt: string;
+      };
+      // check if the user was created in the last 10 minutes
+      // this is a workaround – in the future once we move to use the Account model in the DB
+      // we should use NextAuth's isNewUser flag instead: https://next-auth.js.org/configuration/events#signin
+      const isNewUser = new Date(user.createdAt) > new Date(Date.now() - 10 * 60 * 1000);
```

Comment:
@steven-tey What if we do the opposite here for creating links for existing users. 

Check if theyre not a new user - create a link ect. We should probably add a field to our user object to store referalLinks too. Lmk what you think

Reply:
@sean-brydon that's a great idea – we're thinking of doing something similar on Dub too, where we let users "request a referral link".
- When they do that for the first time, use `dub.links.create` to create a new link and store the linkId under `referralLinkId` field
- All subsequent requests will just copy the link as per their cal.com username (`refer.cal.com/:username`)

We still recommend [setting the `externalId` field when creating a link](https://dub.co/docs/quickstart/nextjs#3a-create-link-with-externalid) though since it gives you the flexibility to easily update the link later on using the `userId` instead of the `referralLinkId`.

Reply:
But with this approach, some of the users' booking pages will not have a valid `refer.cal.com/:username` link, is that okay? cc: @PeerRich 

In that case, it'll default back to the root `refer.cal.com` link, which still leads back to cal.com so it should be fine I think!

Reply:
sure. a fallback is nice, but would they not get a link the moment they log in and not have on?

!hasDubLink && createDubLink()